Realm/Realm.Unity/Editor/Extensions/MethodReferenceExtensions.cs

```diff
@@ -49,4 +49,25 @@ public static MethodReference MakeHostInstanceGeneric(this MethodReference @this
 
         return reference;
     }
+
+    public static bool HasSameSignatureAs(this MethodReference @this, MethodReference other)
+    {
+        if (@this is null || other is null)
+        {
+            return false;
+        }
+
+        return @this.FullName == other.FullName;
+    }
+
+    public static bool ConstructsSameAs(this MethodReference @this, MethodReference other)
```

Comment:
`ConstructsSameAs` is probably not the correct name for this one as it doesn't actually check that the method is a constructor. For the sake of this PR, I would suggest the following changes:

1. In ImportedReferences, instead of constructing a MethodReference to the SyncConfiguration constructor, you can construct a TypeReference for SyncConfiguration.
2. The name of this method can be changed to `ConstructsType(this MethodReference method, TypeReference type)`.
3. The implementation should be changed to `@this.DeclaringType.IsSameAs(type) && @this.Name == ".ctor"`.